### PR TITLE
fix(arns): update event emitter to provide more events and logs while…

### PR DESCRIPTION
**Pull request by dtfiedler**
… loading arns records

Also allows clients to provide their own arns event emitter to handle loaded records

### CODE REVIEW

**Review by atticusofsparta — APPROVED** [review]

File: src/utils/processes.ts [hunk 1]
```diff
@@ -203,6 +217,9 @@ export const fetchAllArNSRecords = async ({
           message: e?.message,
           stack: e?.stack,
         });
+
+        emitter?.emit('arnsError', `Error getting ArNS records: ${e}`);
```
Comment (atticusofsparta) [inline src/utils/processes.ts:221]: maybe we can use the convention of "arns:<event>" here? I have seen it used elsewhere to seperate name spaces.

Comment (dtfiedler) [reply src/utils/processes.ts]: [1d67dfe](https://github.com/ar-io/ar-io-sdk/pull/168/commits/1d67dfe1f7a4766bbce6cf5a55438f33423f660f)

**Review by atticusofsparta — APPROVED** [review]

**Pull request merged**